// @payloadcms/ui/dist/elements/DraggableSortable/index.js
'use client';

import { c as _c } from "react/compiler-runtime";
import { jsx as _jsx } from "react/jsx-runtime";
import { closestCenter, DndContext, KeyboardSensor, MouseSensor, TouchSensor, useDroppable, useSensor, useSensors } from '@dnd-kit/core';
import { SortableContext, sortableKeyboardCoordinates } from '@dnd-kit/sortable';
import React, { useCallback, useId } from 'react';
export const DraggableSortable = props => {
  const $ = _c(20);
  const {
    children,
    className,
    ids,
    onDragEnd,
    onDragStart
  } = props;
  const dndContextID = useId();
  const sortableContextID = useId();
  let t0;
  if ($[0] !== dndContextID) {
    t0 = {
      id: dndContextID
    };
    $[0] = dndContextID;
    $[1] = t0;
  } else {
    t0 = $[1];
  }
  const {
    setNodeRef
  } = useDroppable(t0);
  let t1;
  if ($[2] === Symbol.for("react.memo_cache_sentinel")) {
    t1 = {
      activationConstraint: {
        distance: 5
      }
    };
    $[2] = t1;
  } else {
    t1 = $[2];
  }
  let t2;
  if ($[3] === Symbol.for("react.memo_cache_sentinel")) {
    t2 = {
      activationConstraint: {
        delay: 200,
        tolerance: 5
      }
    };
    $[3] = t2;
  } else {
    t2 = $[3];
  }
  let t3;
  if ($[4] === Symbol.for("react.memo_cache_sentinel")) {
    t3 = {
      coordinateGetter: sortableKeyboardCoordinates
    };
    $[4] = t3;
  } else {
    t3 = $[4];
  }
  const sensors = useSensors(useSensor(MouseSensor, t1), useSensor(TouchSensor, t2), useSensor(KeyboardSensor, t3));
  let t4;
  if ($[5] !== ids || $[6] !== onDragEnd) {
    t4 = event => {
      const {
        active,
        over
      } = event;
      event.activatorEvent.stopPropagation();
      if (!active || !over) {
        return;
      }
      if (typeof onDragEnd === "function") {
        onDragEnd({
          event,
          moveFromIndex: ids.findIndex(_id => _id === active.id),
          moveToIndex: ids.findIndex(_id_0 => _id_0 === over.id)
        });
      }
    };
    $[5] = ids;
    $[6] = onDragEnd;
    $[7] = t4;
  } else {
    t4 = $[7];
  }
  const handleDragEnd = t4;
  let t5;
  if ($[8] !== onDragStart) {
    t5 = event_0 => {
      const {
        active: active_0
      } = event_0;
      if (!active_0) {
        return;
      }
      if (typeof onDragStart === "function") {
        onDragStart({
          id: active_0.id,
          event: event_0
        });
      }
    };
    $[8] = onDragStart;
    $[9] = t5;
  } else {
    t5 = $[9];
  }
  const handleDragStart = t5;
  let t6;
  if ($[10] !== children || $[11] !== className || $[12] !== dndContextID || $[13] !== handleDragEnd || $[14] !== handleDragStart || $[15] !== ids || $[16] !== sensors || $[17] !== setNodeRef || $[18] !== sortableContextID) {
    t6 = _jsx(DndContext, {
      collisionDetection: closestCenter,
      id: dndContextID,
      onDragEnd: handleDragEnd,
      onDragStart: handleDragStart,
      sensors,
      children: _jsx(SortableContext, {
        id: sortableContextID,
        items: ids,
        children: _jsx("div", {
          className,
          ref: setNodeRef,
          children
        })
      })
    });
    $[10] = children;
    $[11] = className;
    $[12] = dndContextID;
    $[13] = handleDragEnd;
    $[14] = handleDragStart;
    $[15] = ids;
    $[16] = sensors;
    $[17] = setNodeRef;
    $[18] = sortableContextID;
    $[19] = t6;
  } else {
    t6 = $[19];
  }
  return t6;
};
