// @payloadcms/ui/dist/elements/QueryPresets/fields/WhereField/index.js
'use client';

import { c as _c } from "react/compiler-runtime";
import { jsx as _jsx, jsxs as _jsxs } from "react/jsx-runtime";
import React from 'react';
import { FieldLabel } from '../../../../fields/FieldLabel/index.js';
import { useField } from '../../../../forms/useField/index.js';
import { WhereBuilder } from '../../../WhereBuilder/index.js';
import './index.scss';
export const QueryPresetsWhereField = t0 => {
  const $ = _c(14);
  const {
    field: t1
  } = t0;
  const {
    label,
    required
  } = t1;
  const {
    path,
    setValue,
    value
  } = useField();
  let t2;
  if ($[0] === Symbol.for("react.memo_cache_sentinel")) {
    t2 = {
      path: "relatedCollection"
    };
    $[0] = t2;
  } else {
    t2 = $[0];
  }
  const relatedCollectionField = useField(t2);
  const relatedCollection = relatedCollectionField.value;
  if (!relatedCollection) {
    let t3;
    if ($[1] !== label || $[2] !== path || $[3] !== required) {
      t3 = _jsxs("div", {
        className: "field-type query-preset-where-field",
        children: [_jsx(FieldLabel, {
          as: "h3",
          label,
          path,
          required
        }), _jsx("p", {
          className: "query-preset-where-field__hint",
          children: "Select the related collection to configure filters."
        })]
      });
      $[1] = label;
      $[2] = path;
      $[3] = required;
      $[4] = t3;
    } else {
      t3 = $[4];
    }
    return t3;
  }
  let t3;
  if ($[5] !== label || $[6] !== path || $[7] !== relatedCollection || $[8] !== required || $[9] !== setValue || $[10] !== value) {
    let t4;
    if ($[12] !== setValue) {
      t4 = where => setValue(where);
      $[12] = setValue;
      $[13] = t4;
    } else {
      t4 = $[13];
    }
    t3 = _jsxs("div", {
      className: "field-type query-preset-where-field",
      children: [_jsx(FieldLabel, {
        as: "h3",
        label,
        path,
        required
      }), _jsx(WhereBuilder, {
        collectionSlug: relatedCollection,
        onChange: t4,
        value: value ?? undefined
      })]
    });
    $[5] = label;
    $[6] = path;
    $[7] = relatedCollection;
    $[8] = required;
    $[9] = setValue;
    $[10] = value;
    $[11] = t3;
  } else {
    t3 = $[11];
  }
  return t3;
};
