// @payloadcms/ui/dist/elements/WhereBuilder/Condition/Date/index.js
'use client';

import { jsx as _jsx } from "react/jsx-runtime";
import { getTranslation } from '@payloadcms/translations';
import React from 'react';
import { useConfig } from '../../../../providers/Config/index.js';
import { useTranslation } from '../../../../providers/Translation/index.js';
import { DatePickerField } from '../../../DatePicker/index.js';
const baseClass = 'condition-value-date';
export const DateFilter = t0 => {
  const {
    disabled,
    field: t1,
    onChange,
    value
  } = t0;
  const {
    admin
  } = t1;
  const {
    date
  } = admin || {};
  const {
    config: t2
  } = useConfig();
  const {
    admin: t3
  } = t2;
  const {
    dateFormat: dateFormatFromConfig
  } = t3;
  const {
    i18n,
    t
  } = useTranslation();
  const displayFormat = date?.displayFormat || dateFormatFromConfig;
  return _jsx("div", {
    className: baseClass,
    children: _jsx(DatePickerField, {
      ...date,
      displayFormat,
      onChange,
      placeholder: getTranslation(admin.placeholder, i18n) || t("general:enterAValue"),
      readOnly: disabled,
      value
    })
  });
};
