// @payloadcms/ui/dist/fields/Slug/index.js
'use client';

import { c as _c } from "react/compiler-runtime";
import { jsx as _jsx, jsxs as _jsxs } from "react/jsx-runtime";
import React, { useCallback, useState } from 'react';
import { Button } from '../../elements/Button/index.js';
import { useForm } from '../../forms/Form/index.js';
import { useField } from '../../forms/useField/index.js';
import { useDocumentInfo } from '../../providers/DocumentInfo/index.js';
import { useServerFunctions } from '../../providers/ServerFunctions/index.js';
import { useTranslation } from '../../providers/Translation/index.js';
import { FieldLabel } from '../FieldLabel/index.js';
import { TextInput } from '../Text/index.js';
import './index.scss';
/**
 * @experimental This component is experimental and may change or be removed in the future. Use at your own risk.
 */
export const SlugField = t0 => {
  const $ = _c(26);
  const {
    field,
    path,
    useAsSlug
  } = t0;
  const {
    admin,
    label
  } = field;
  let t1;
  if ($[0] !== admin) {
    t1 = admin || {};
    $[0] = admin;
    $[1] = t1;
  } else {
    t1 = $[1];
  }
  const {
    readOnly: readOnlyFromProps
  } = t1;
  const {
    t
  } = useTranslation();
  const {
    collectionSlug,
    globalSlug
  } = useDocumentInfo();
  const {
    slugify
  } = useServerFunctions();
  const t2 = path || field.name;
  let t3;
  if ($[2] !== t2) {
    t3 = {
      path: t2
    };
    $[2] = t2;
    $[3] = t3;
  } else {
    t3 = $[3];
  }
  const {
    setValue,
    value
  } = useField(t3);
  const {
    getData,
    getDataByPath
  } = useForm();
  const [isLocked, setIsLocked] = useState(true);
  let t4;
  if ($[4] !== collectionSlug || $[5] !== getData || $[6] !== getDataByPath || $[7] !== globalSlug || $[8] !== path || $[9] !== setValue || $[10] !== slugify || $[11] !== useAsSlug || $[12] !== value) {
    t4 = async e => {
      e.preventDefault();
      const valueToSlugify = getDataByPath(useAsSlug);
      const formattedSlug = await slugify({
        collectionSlug,
        data: getData(),
        globalSlug,
        path,
        valueToSlugify
      });
      if (formattedSlug === null || formattedSlug === undefined) {
        setValue("");
        return;
      }
      if (value !== formattedSlug) {
        setValue(formattedSlug);
      }
    };
    $[4] = collectionSlug;
    $[5] = getData;
    $[6] = getDataByPath;
    $[7] = globalSlug;
    $[8] = path;
    $[9] = setValue;
    $[10] = slugify;
    $[11] = useAsSlug;
    $[12] = value;
    $[13] = t4;
  } else {
    t4 = $[13];
  }
  const handleGenerate = t4;
  let t5;
  if ($[14] === Symbol.for("react.memo_cache_sentinel")) {
    t5 = e_0 => {
      e_0.preventDefault();
      setIsLocked(_temp);
    };
    $[14] = t5;
  } else {
    t5 = $[14];
  }
  const toggleLock = t5;
  const t6 = `field-${path}`;
  let t7;
  if ($[15] !== field.name || $[16] !== handleGenerate || $[17] !== isLocked || $[18] !== label || $[19] !== path || $[20] !== readOnlyFromProps || $[21] !== setValue || $[22] !== t || $[23] !== t6 || $[24] !== value) {
    t7 = _jsxs("div", {
      className: "field-type slug-field-component",
      children: [_jsxs("div", {
        className: "label-wrapper",
        children: [_jsx(FieldLabel, {
          htmlFor: t6,
          label
        }), !readOnlyFromProps && !isLocked && _jsx(Button, {
          buttonStyle: "none",
          className: "lock-button",
          onClick: handleGenerate,
          children: t("authentication:generate")
        }), !readOnlyFromProps && _jsx(Button, {
          buttonStyle: "none",
          className: "lock-button",
          onClick: toggleLock,
          children: isLocked ? t("general:unlock") : t("general:lock")
        })]
      }), _jsx(TextInput, {
        onChange: setValue,
        path: path || field.name,
        readOnly: Boolean(readOnlyFromProps || isLocked),
        value
      })]
    });
    $[15] = field.name;
    $[16] = handleGenerate;
    $[17] = isLocked;
    $[18] = label;
    $[19] = path;
    $[20] = readOnlyFromProps;
    $[21] = setValue;
    $[22] = t;
    $[23] = t6;
    $[24] = value;
    $[25] = t7;
  } else {
    t7 = $[25];
  }
  return t7;
};
function _temp(prev) {
  return !prev;
}
