// @payloadcms/ui/dist/elements/QueryPresets/fields/GroupByField/index.js
'use client';

import { jsx as _jsx, jsxs as _jsxs } from "react/jsx-runtime";
import React, { useMemo } from 'react';
import { FieldLabel } from '../../../../fields/FieldLabel/index.js';
import { useField } from '../../../../forms/useField/index.js';
import { useConfig } from '../../../../providers/Config/index.js';
import { GroupByBuilder } from '../../../GroupByBuilder/index.js';
import './index.scss';
export const QueryPresetsGroupByField = ({
  field: {
    label,
    required
  }
}) => {
  const {
    path,
    setValue,
    value
  } = useField();
  const relatedCollectionField = useField({
    path: 'relatedCollection'
  });
  const relatedCollection = relatedCollectionField.value;
  const {
    config
  } = useConfig();
  const collectionConfig = useMemo(() => {
    if (!relatedCollection) {
      return null;
    }
    return config.collections?.find(col => col.slug === relatedCollection);
  }, [relatedCollection, config.collections]);
  const fields = collectionConfig?.fields ?? [];
  if (!relatedCollection) {
    return /*#__PURE__*/_jsxs("div", {
      className: "field-type query-preset-group-by-field",
      children: [/*#__PURE__*/_jsx(FieldLabel, {
        as: "h3",
        label: label,
        path: path,
        required: required
      }), /*#__PURE__*/_jsx("p", {
        className: "query-preset-group-by-field__hint",
        children: "Select the related collection to configure group by."
      })]
    });
  }
  return /*#__PURE__*/_jsxs("div", {
    className: "field-type query-preset-group-by-field",
    children: [/*#__PURE__*/_jsx(FieldLabel, {
      as: "h3",
      label: label,
      path: path,
      required: required
    }), /*#__PURE__*/_jsx(GroupByBuilder, {
      collectionSlug: relatedCollection,
      fields: fields,
      onChange: groupBy => setValue(groupBy || null),
      value: value ?? undefined
    })]
  });
};
