// @payloadcms/ui/dist/elements/GroupByBuilder/index.js
'use client';

import { jsx as _jsx, jsxs as _jsxs } from "react/jsx-runtime";
import './index.scss';
import React, { useMemo } from 'react';
import { SelectInput } from '../../fields/Select/Input.js';
import { useAuth } from '../../providers/Auth/index.js';
import { useListQuery } from '../../providers/ListQuery/index.js';
import { useTranslation } from '../../providers/Translation/index.js';
import { reduceFieldsToOptions } from '../../utilities/reduceFieldsToOptions.js';
import { ReactSelect } from '../ReactSelect/index.js';
const baseClass = 'group-by-builder';
/**
 * Note: Some fields are already omitted from the list of fields:
 * - fields with nested field, e.g. `tabs`, `groups`, etc.
 * - fields that don't affect data, i.e. `row`, `collapsible`, `ui`, etc.
 * So we don't technically need to omit them here, but do anyway.
 * But some remaining fields still need an additional check, e.g. `richText`, etc.
 */
const supportedFieldTypes = ['text', 'textarea', 'number', 'select', 'relationship', 'date', 'checkbox', 'radio', 'email', 'number', 'upload'];
export const GroupByBuilder = ({
  collectionSlug,
  fields,
  onChange,
  value: valueProp
}) => {
  const {
    i18n,
    t
  } = useTranslation();
  const {
    permissions
  } = useAuth();
  const listQuery = useListQuery();
  const isFormMode = typeof onChange === 'function';
  const groupByRaw = isFormMode ? valueProp : listQuery.query?.groupBy;
  const groupByFieldName = groupByRaw?.replace(/^-/, '');
  const fieldPermissions = permissions?.collections?.[collectionSlug]?.fields;
  const reducedFields = useMemo(() => reduceFieldsToOptions({
    fieldPermissions,
    fields,
    i18n
  }), [fields, fieldPermissions, i18n]);
  const groupByField = reducedFields.find(field => field.value === groupByFieldName);
  const handleFieldChange = useMemo(() => {
    if (isFormMode) {
      return v => {
        const value = v === null ? undefined : v.value;
        const newGroupBy = value ? groupByRaw?.startsWith('-') ? `-${value}` : value : '';
        onChange?.(newGroupBy);
      };
    }
    return v_0 => {
      void (async () => {
        if (typeof listQuery.refineListData !== 'function') {
          return;
        }
        const value_0 = v_0 === null ? undefined : v_0.value;
        if (v_0 === null) {
          await listQuery.refineListData({
            groupBy: '',
            page: 1
          });
        } else {
          await listQuery.refineListData({
            groupBy: value_0 ? listQuery.query?.groupBy?.startsWith('-') ? `-${value_0}` : value_0 : undefined,
            page: 1
          });
        }
      })();
    };
  }, [isFormMode, groupByRaw, listQuery, onChange]);
  const handleClear = useMemo(() => {
    if (isFormMode) {
      return () => onChange?.('');
    }
    return () => {
      void (async () => {
        if (typeof listQuery.refineListData === 'function') {
          await listQuery.refineListData({
            groupBy: ''
          });
        }
      })();
    };
  }, [isFormMode, listQuery, onChange]);
  const handleDirectionChange = useMemo(() => {
    if (isFormMode) {
      return ({
        value: value_1
      }) => {
        if (!groupByFieldName) {
          return;
        }
        onChange?.(value_1 === 'asc' ? groupByFieldName : `-${groupByFieldName}`);
      };
    }
    return ({
      value: value_2
    }) => {
      void (async () => {
        if (!groupByFieldName || typeof listQuery.refineListData !== 'function') {
          return;
        }
        await listQuery.refineListData({
          groupBy: value_2 === 'asc' ? groupByFieldName : `-${groupByFieldName}`,
          page: 1
        });
      })();
    };
  }, [groupByFieldName, isFormMode, listQuery, onChange]);
  const directionValue = !groupByRaw || typeof groupByRaw !== 'string' ? 'asc' : groupByRaw.startsWith('-') ? 'desc' : 'asc';
  return /*#__PURE__*/_jsxs("div", {
    className: baseClass,
    children: [/*#__PURE__*/_jsxs("div", {
      className: `${baseClass}__header`,
      children: [/*#__PURE__*/_jsx("p", {
        children: t('general:groupByLabel', {
          label: ''
        })
      }), groupByRaw && /*#__PURE__*/_jsx("button", {
        className: `${baseClass}__clear-button`,
        id: "group-by--reset",
        onClick: () => void handleClear(),
        type: "button",
        children: t('general:clear')
      })]
    }), /*#__PURE__*/_jsxs("div", {
      className: `${baseClass}__inputs`,
      children: [/*#__PURE__*/_jsx(ReactSelect, {
        filterOption: (option, inputValue) => (option?.data?.plainTextLabel || option.label).toLowerCase().includes(inputValue.toLowerCase()),
        id: "group-by--field-select",
        isClearable: true,
        isMulti: false,
        onChange: handleFieldChange,
        options: reducedFields.filter(field_0 => !field_0.field.admin?.disableGroupBy && field_0.value !== 'id' && supportedFieldTypes.includes(field_0.field.type)),
        value: {
          label: groupByField?.label || t('general:selectValue'),
          value: groupByFieldName || ''
        }
      }), /*#__PURE__*/_jsx(SelectInput, {
        id: "group-by--sort",
        isClearable: false,
        name: "direction",
        onChange: handleDirectionChange,
        options: [{
          label: t('general:ascending'),
          value: 'asc'
        }, {
          label: t('general:descending'),
          value: 'desc'
        }],
        path: "direction",
        readOnly: !groupByFieldName,
        value: directionValue
      })]
    })]
  });
};
