// @payloadcms/ui/dist/elements/DocumentStaleData/index.js
'use client';

import { c as _c } from "react/compiler-runtime";
import { jsx as _jsx, jsxs as _jsxs } from "react/jsx-runtime";
import React, { useEffect } from 'react';
import { useRouteCache } from '../../providers/RouteCache/index.js';
import { useTranslation } from '../../providers/Translation/index.js';
import { Button } from '../Button/index.js';
import { Modal, useModal } from '../Modal/index.js';
import './index.scss';
const modalSlug = 'document-stale-data';
const baseClass = 'document-stale-data';
export const DocumentStaleData = t0 => {
  const $ = _c(14);
  const {
    isActive,
    onReload
  } = t0;
  const {
    closeModal,
    openModal
  } = useModal();
  const {
    clearRouteCache
  } = useRouteCache();
  const {
    t
  } = useTranslation();
  let t1;
  let t2;
  if ($[0] !== closeModal || $[1] !== isActive || $[2] !== openModal) {
    t1 = () => {
      if (isActive) {
        openModal(modalSlug);
      } else {
        closeModal(modalSlug);
      }
    };
    t2 = [isActive, openModal, closeModal];
    $[0] = closeModal;
    $[1] = isActive;
    $[2] = openModal;
    $[3] = t1;
    $[4] = t2;
  } else {
    t1 = $[3];
    t2 = $[4];
  }
  useEffect(t1, t2);
  let t3;
  if ($[5] !== clearRouteCache || $[6] !== closeModal || $[7] !== onReload || $[8] !== t) {
    let t4;
    if ($[10] !== clearRouteCache || $[11] !== closeModal || $[12] !== onReload) {
      t4 = async () => {
        closeModal(modalSlug);
        clearRouteCache();
        await onReload();
      };
      $[10] = clearRouteCache;
      $[11] = closeModal;
      $[12] = onReload;
      $[13] = t4;
    } else {
      t4 = $[13];
    }
    t3 = _jsx(Modal, {
      className: baseClass,
      closeOnBlur: false,
      slug: modalSlug,
      children: _jsxs("div", {
        className: `${baseClass}__wrapper`,
        children: [_jsxs("div", {
          className: `${baseClass}__content`,
          children: [_jsx("h1", {
            children: t("general:documentModified")
          }), _jsx("p", {
            children: t("general:documentOutOfDate")
          })]
        }), _jsx("div", {
          className: `${baseClass}__controls`,
          children: _jsx(Button, {
            buttonStyle: "primary",
            id: `${modalSlug}-reload`,
            margin: false,
            onClick: t4,
            size: "medium",
            children: t("general:reloadDocument")
          })
        })]
      })
    });
    $[5] = clearRouteCache;
    $[6] = closeModal;
    $[7] = onReload;
    $[8] = t;
    $[9] = t3;
  } else {
    t3 = $[9];
  }
  return t3;
};
