// @payloadcms/ui/dist/elements/ListHeader/TitleActions/ListBulkUploadButton.js
'use client';

import { c as _c } from "react/compiler-runtime";
import { jsx as _jsx } from "react/jsx-runtime";
import { useModal } from '@faceless-ui/modal';
import { useRouter } from 'next/navigation.js';
import React from 'react';
import { useBulkUpload } from '../../../elements/BulkUpload/index.js';
import { useFolder } from '../../../providers/Folders/index.js';
import { useTranslation } from '../../../providers/Translation/index.js';
import { Button } from '../../Button/index.js';
export function ListBulkUploadButton(t0) {
  const $ = _c(14);
  const {
    collectionSlug,
    hasCreatePermission,
    isBulkUploadEnabled,
    onBulkUploadSuccess,
    openBulkUpload: openBulkUploadFromProps
  } = t0;
  const {
    drawerSlug: bulkUploadDrawerSlug,
    setCollectionSlug,
    setFolderID,
    setOnSuccess
  } = useBulkUpload();
  const {
    folderID
  } = useFolder();
  const {
    t
  } = useTranslation();
  const {
    openModal
  } = useModal();
  const router = useRouter();
  let t1;
  if ($[0] !== bulkUploadDrawerSlug || $[1] !== collectionSlug || $[2] !== folderID || $[3] !== onBulkUploadSuccess || $[4] !== openBulkUploadFromProps || $[5] !== openModal || $[6] !== router || $[7] !== setCollectionSlug || $[8] !== setFolderID || $[9] !== setOnSuccess) {
    t1 = () => {
      if (typeof openBulkUploadFromProps === "function") {
        openBulkUploadFromProps();
      } else {
        setCollectionSlug(collectionSlug);
        setFolderID(folderID);
        openModal(bulkUploadDrawerSlug);
        setOnSuccess(() => {
          if (typeof onBulkUploadSuccess === "function") {
            onBulkUploadSuccess();
          } else {
            router.refresh();
          }
        });
      }
    };
    $[0] = bulkUploadDrawerSlug;
    $[1] = collectionSlug;
    $[2] = folderID;
    $[3] = onBulkUploadSuccess;
    $[4] = openBulkUploadFromProps;
    $[5] = openModal;
    $[6] = router;
    $[7] = setCollectionSlug;
    $[8] = setFolderID;
    $[9] = setOnSuccess;
    $[10] = t1;
  } else {
    t1 = $[10];
  }
  const openBulkUpload = t1;
  if (!hasCreatePermission || !isBulkUploadEnabled) {
    return null;
  }
  let t2;
  if ($[11] !== openBulkUpload || $[12] !== t) {
    t2 = _jsx(Button, {
      "aria-label": t("upload:bulkUpload"),
      buttonStyle: "pill",
      onClick: openBulkUpload,
      size: "small",
      children: t("upload:bulkUpload")
    }, "bulk-upload-button");
    $[11] = openBulkUpload;
    $[12] = t;
    $[13] = t2;
  } else {
    t2 = $[13];
  }
  return t2;
}
