// @payloadcms/ui/dist/elements/Thumbnail/index.js
'use client';

import { jsx as _jsx, jsxs as _jsxs } from "react/jsx-runtime";
import React from 'react';
import './index.scss';
const baseClass = 'thumbnail';
import { File } from '../../graphics/File/index.js';
import { appendCacheTag } from '../../utilities/appendCacheTag.js';
import { ShimmerEffect } from '../ShimmerEffect/index.js';
export const Thumbnail = props => {
  const {
    className = '',
    doc: {
      filename
    } = {},
    fileSrc,
    height,
    imageCacheTag,
    size,
    width
  } = props;
  const [fileExists, setFileExists] = React.useState(undefined);
  const classNames = [baseClass, `${baseClass}--size-${size || 'medium'}`, className].join(' ');
  const src = React.useMemo(() => fileSrc ? appendCacheTag(fileSrc, imageCacheTag) : null, [fileSrc, imageCacheTag]);
  React.useEffect(() => {
    if (!src) {
      setFileExists(false);
      return;
    }
    setFileExists(undefined);
    const img = new Image();
    img.src = src;
    img.onload = () => {
      setFileExists(true);
    };
    img.onerror = () => {
      setFileExists(false);
    };
  }, [src]);
  return /*#__PURE__*/_jsxs("div", {
    className: classNames,
    children: [fileExists === undefined && /*#__PURE__*/_jsx(ShimmerEffect, {
      height: "100%"
    }), fileExists && /*#__PURE__*/_jsx("img", {
      alt: filename,
      height: height,
      src: src,
      width: width
    }), fileExists === false && /*#__PURE__*/_jsx(File, {})]
  });
};
export function ThumbnailComponent(props) {
  const {
    alt,
    className = '',
    filename,
    fileSrc,
    imageCacheTag,
    size
  } = props;
  const [fileExists, setFileExists] = React.useState(undefined);
  const classNames = [baseClass, `${baseClass}--size-${size || 'medium'}`, className].join(' ');
  const src = React.useMemo(() => fileSrc ? appendCacheTag(fileSrc, imageCacheTag) : null, [fileSrc, imageCacheTag]);
  React.useEffect(() => {
    if (!src) {
      setFileExists(false);
      return;
    }
    setFileExists(undefined);
    const img = new Image();
    img.src = src;
    img.onload = () => {
      setFileExists(true);
    };
    img.onerror = () => {
      setFileExists(false);
    };
  }, [src]);
  return /*#__PURE__*/_jsxs("div", {
    className: classNames,
    children: [fileExists === undefined && /*#__PURE__*/_jsx(ShimmerEffect, {
      height: "100%"
    }), fileExists && /*#__PURE__*/_jsx("img", {
      alt: alt || filename,
      src: src
    }), fileExists === false && /*#__PURE__*/_jsx(File, {})]
  });
}
